momento-sdk/src/main/java/momento/sdk/auth/StringCredentialProvider.java

```diff
@@ -21,8 +39,25 @@ public class StringCredentialProvider implements CredentialProvider {
    * @param authToken a Momento JWT.
    */
   public StringCredentialProvider(@Nonnull String authToken) {
-    this.authToken = authToken;
+    TokenAndEndpoints data;
+    try {
+      data = processV1Token(authToken);
+    } catch (IllegalArgumentException iae) {
+      // if base64 decoding fails, fall back to processing legacy token
+      if (iae.getMessage().contains("base64")) {
+        data = processLegacyToken(authToken);
+      } else {
+        throw iae;
```

Comment:
Can you wrap this in an InvalidArgumentException so we only throw SdkExceptions?

Reply:
Fixed in 7e0e8df